transport/socks5/socks5.go

```diff
@@ -195,17 +195,29 @@ func ClientHandshake(rw io.ReadWriter, addr Addr, command Command, user *User) (
 			return nil, errors.New("auth required")
 		}
 
-		authMsgLen := 1 + 1 + len(user.Username) + 1 + len(user.Password)
-		if authMsgLen > MaxAuthLen {
-			return nil, errors.New("auth message too long")
+		uLen := len(user.Username)
+		pLen := len(user.Password)
+
+		// Both ULEN and PLEN are limited to the range [1, 255].
+		switch {
+		case uLen == 0:
```

Comment:
I think here we can simply check if username/password is missing or too long, no need to check them separately.

Reply:
Fixed, I hope.

Reply:
Can be this:

```go
if uLen == 0 || pLen == 0 {
	return nil, errors.New("auth username/password empty")
} else if uLen > MaxAuthLen || pLen > MaxAuthLen {
	return nil, errors.New("auth username/password too long")
}
```

Reply:
Done.